### PR TITLE
Pattern persistance

### PR DESCRIPTION
**Pull request by connorlay**
Adds `Pattern` and `StitchPatternRelation` models. A `Pattern` has many `Stitch` models through the `StitchPatternRelation` model.

On first launch, the database is populated with the default stitches, as well as a default pattern. The first pattern is then pulled from the database and passed to `PatternDetailFragment`.

I also updated some of the resource names to be more consistent.

### CODE REVIEW

File: app/src/main/java/com/connorlay/knitgrid/models/Pattern.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.List;
+
+/**
+ * Created by connorlay on 3/17/16.
+ */
+public class Pattern extends SugarRecord implements Parcelable {
+    private String name;
+    private int rows;
+    private int cols;
+    private boolean alternates;
```
Comment (Magisus) [inline app/src/main/java/com/connorlay/knitgrid/models/Pattern.java]: I don't like this name. I would use alternating, and then use 'alternating' as well for both the getter 'isAlternating' and the setter 'setAlternating'.

**Comment by Magisus** [comment]
I think this is a good model!

File: app/src/main/java/com/connorlay/knitgrid/models/Pattern.java (hunk 1 above)
Comment (connorlay) [inline app/src/main/java/com/connorlay/knitgrid/models/Pattern.java]: Yeah that makes sense. I'll change that now.

**Comment by connorlay** [comment]
Ok if that looks good I'll squash it. Also I don't think SugarORM has built in database level constraints... but I think it's ok since we can still perform validations at the model level when we need to.

**Comment by Magisus** [comment]
Alright sounds good. Squash it up and I'll merge!

**Comment by Magisus** [comment]
Apparently there are conflicts now?

**Pull request merged**